scripts/smart_version_resolver.py

```
@@ -0,0 +1,415 @@
+        if requested_tuple == suggested_tuple:
+            # Perfect match
+            resolved_version = requested_version
+            reasoning = f"Requested version matches analysis: {bump_type} bump appropriate"
+        elif requested_tuple > suggested_tuple:
+            # Requested is higher - might be future planning
+            if requested_tuple[0] > current_tuple[0]:  # Major bump
+                resolved_version = requested_version if impact_scores['major'] > 0 else suggested_version
+                reasoning = f"Major version requested but commits suggest {bump_type} bump"
+            else:
+                resolved_version = requested_version
+                reasoning = f"Accepting higher requested version for future planning"
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)